test_data/makedata_suites/402_views.js

```diff
@@ -0,0 +1,470 @@
+
+  let getMetricCluster = function (name) {
+    let headers = {};
+    headers['accept'] = 'application/json';
+    headers["Authorization"] = `Bearer ${jwt_key}`;
+    let clusterHealth = arango.GET_RAW("/_admin/cluster/health", headers)["parsedBody"]["Health"];
+
+    let serversId = [];
+    for (let [key, value] of Object.entries(clusterHealth)) {
+      if (value.Role.toLowerCase() == "dbserver") {
+        serversId.push(key);
+      }
+    }
+
+    let value = 0;
+    for (let i = 0; i < serversId.length; i++) {
+      value += getMetricByName(name, `?serverId=${serversId[i]}`);
+    }
+
+    return value;
+  };
+
+  let getMetric = function (name, options) {
+    generateJWT(options);
+    if (isCluster) {
+      return getMetricCluster(name);
+    } else {
+      return getMetricSingle(name);
+    }
+  };
+
+  let isCacheSizeSupported = function (version) {
+    return (semver.eq(version, "3.9.5") || semver.gte(version, "3.10.2"));
+  };
+  return {
+    isSupported: function (version, oldVersion, enterprise, cluster) {
+      return semver.gte(version, '3.9.5');
+    },
+    makeData: function (options, isCluster, isEnterprise, dbCount, loopCount) {
+      // All items created must contain dbCount and loopCount
+      print(`making data 402 ${dbCount} ${loopCount}`);
+
+      // create analyzer with 'norm' feature
+      const analyzers = require("@arangodb/analyzers");
+      analyzers.save("AqlAnalyzerHash", "aql", { queryString: "return to_hex(to_string(@param))" }, ["frequency", "norm", "position"])
+
+      // create views for testing
+      progress('createViewCache');
+      let viewNameCache = `viewCache_${loopCount}`;
+      let viewCache = createSafe(viewNameCache,
+        viewNameCache => {
+          return db._createView(viewNameCache, "arangosearch", { "storedValues": [{ "fields": ["animal", "name"], "cache": true }] });
+        }, viewNameCache => {
+          return db._view(viewNameCache);
+        }
+      );
+
+      progress('createViewNoCache');
+      let viewNameNoCache = `viewNoCache_${loopCount}`;
+      let viewNoCache = createSafe(viewNameNoCache,
+        viewNameNoCache => {
+          return db._createView(viewNameNoCache, "arangosearch", { "storedValues": [{ "fields": ["animal", "name"], "cache": false }] });
+        }, viewNameNoCache => {
+          return db._view(viewNameNoCache);
+        }
+      );
+
+      let currVersion = db._version();
+      let cacheSizeSupported = isCacheSizeSupported(currVersion);
+
+      let cacheSize = 0;
+      let prevCacheSize = cacheSize;
+
+      if (cacheSizeSupported) {
+        cacheSize = getMetric("arangodb_search_columns_cache_size", options);
+        if (cacheSize != 0) {
+          throw new Error("initial cache size is not 0");
+        }
+        prevCacheSize = cacheSize;
+      }
+
+      for (let i = 0; i < links.length; i++) {
+        // create collection for each testing link
+        let collectionName = `collectionCache${i}_${loopCount}`;
```

Comment:
I think it would be better to have `collectionCache${i}_` as attribute of `link[i]` with a descriptive name to the test (plus the loopCount for sure)

I think that then `links.forEach( link => {...})` could be used